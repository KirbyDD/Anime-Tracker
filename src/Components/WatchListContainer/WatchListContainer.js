import React from 'react'
import {connect} from 'react-redux';
import AnimeSnippet from '../AnimeSnippet/AnimeSnippet';

const WatchListContainer = props => {
  let list = props.followedShows.map(anime => {
      return <AnimeSnippet {...anime} isFav={true} />   
  })
  return (
    <section>
      {list}
    </section>
  )
}

const mapStateToProps = store => ({
  ...store
})

export default connect(mapStateToProps)(WatchListContainer)