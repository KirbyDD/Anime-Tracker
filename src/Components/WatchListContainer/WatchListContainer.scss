.watchlist {
  display: flex;
  flex-wrap: wrap;
  padding: 50px;
}
