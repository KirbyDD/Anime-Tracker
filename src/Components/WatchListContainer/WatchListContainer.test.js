import React from "react";
import { shallow } from "enzyme";
import { WatchListContainer, mapStateToProps } from "./WatchListContainer";

let anime = [
  {
    name: "anime1",
    id: 1
  },
  {
    name: "anime2",
    id: 2
  },
  {
    name: "anime3",
    id: 3
  }
];

let props = {
  id: 1,
  synopsis: "This is the synopsis",
  enTitle: "Life of TronKat",
  jpTitle: "Tronkat no Life desu",
  episodes: 27,
  posterImage: {
    large: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  },
  coverImage: {
    small: "https://i.imgur.com/vihivug.jpg",
    tiny: null
  },
  rating: "99.9",
  startDate: "1990-01-26",
  endDate: "Still Playing",
  ageRating: "17+"
};

let followedShows = [
  {
    id: 1,
    synopsis: "This is the synopsis",
    enTitle: "Life of TronKat",
    jpTitle: "Tronkat no Life desu",
    episodes: 27,
    posterImage: {
      large: "https://i.imgur.com/vihivug.jpg",
      tiny: null
    },
    coverImage: {
      small: "https://i.imgur.com/vihivug.jpg",
      tiny: null
    },
    rating: "99.9",
    startDate: "1990-01-26",
    endDate: "Still Playing",
    ageRating: "17+"
  }
];


describe("Watch List Container", () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(<WatchListContainer followedShows={anime} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return object with anime array', () => {
      let mockState = {
        anime: props,
        followedShows
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(mockState)
    })
  })
});
