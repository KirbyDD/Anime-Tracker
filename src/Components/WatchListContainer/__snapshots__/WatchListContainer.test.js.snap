// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch List Container should match snapshot 1`] = `
<section
  className="watchlist"
>
  <AnimeSnippet
    id={1}
    isFav={true}
    key="1"
    name="anime1"
  />
  <AnimeSnippet
    id={2}
    isFav={true}
    key="2"
    name="anime2"
  />
  <AnimeSnippet
    id={3}
    isFav={true}
    key="3"
    name="anime3"
  />
</section>
`;
